server/src/main/java/org/opensearch/index/search/comparators/HalfFloatComparator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.search.comparators;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.sandbox.document.HalfFloatPoint;
+import org.apache.lucene.search.LeafFieldComparator;
+import org.apache.lucene.search.comparators.NumericComparator;
+
+import java.io.IOException;
+
+/**
+ * The comparator for unsigned half_float numeric type.
```

Comment:
I think those are signed, right?

```suggestion
 * The comparator for half_float numeric type.
```